### PR TITLE
[runtime] Fix the signature of the AppDomain:DoUnhandledException () icall.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/metadata/icall-def.h [hunk 1]
```diff
@@ -176,7 +176,7 @@ HANDLES(KPAIR_5, "_ProtectUser", ves_icall_Mono_Security_Cryptography_KeyPairPer
 #endif /* !PLATFORM_RO_FS */
 
 ICALL_TYPE(APPDOM, "System.AppDomain", APPDOM_23)
-HANDLES(APPDOM_23, "DoUnhandledException", ves_icall_System_AppDomain_DoUnhandledException, void, 1, (MonoException))
+HANDLES(APPDOM_23, "DoUnhandledException", ves_icall_System_AppDomain_DoUnhandledException, void, 2, (MonoAppDomain, MonoException))
```
Comment (lambdageek) [inline mono/metadata/icall-def.h]: Ouch.  Good catch - it's non-`static`

https://github.com/mono/mono/blob/1af992a5ffa46e20dd61a64b6dcecef0edb5c459/mcs/class/corlib/System/AppDomain.cs#L1405

**Comment by lambdageek** [comment]
@monojenkins backport to 2019-04
@monojenkins backport to 2019-02

**Comment by lambdageek** [comment]
This function was first changed to use coop handles back in 0256888233cf0d94443689f0abb508af11411033 and it's been wrong since then (back to 2017-06).

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by lewing** [comment]
@monojenkins build failed

**Pull request merged**